core/chains/evm/logpoller/log_poller.go

```diff
@@ -121,6 +122,17 @@ type logPoller struct {
 	wg             sync.WaitGroup
 }
 
+type Opts struct {
+	PollPeriod               time.Duration
+	UseFinalityTag           bool
+	FinalityDepth            int64
+	BackfillBatchSize        int64
+	RpcBatchSize             int64
+	KeepFinalizedBlocksDepth int64
+	BackupPollerBlockDelay   int64
+	LogPrunePageSize         int64
```

Comment:
neat pick: any reason to use `int64` instead of `uint64`? I think there's no case in which we should allow passing negative values for any of the options?

Reply:
That's true, but the native type for block numbers on Ethereum is `int64` which makes it much more convenient and less error prone to use that rather than `uint64`.
It's the same issue with `RpcBatchSize` and `KeepFinalizedBlocksDepth` defined just above it.  I've thought of making them into `uint64`'s before, and even tried doing it once, but because we have to do arithmetic on all of these involving actual block numbers, it won't compile unless you cast them to `int64` when using them anyway.  For example, in this line:

```
backupStartBlock := mathutil.Min(lastProcessed.FinalizedBlockNumber-1, lastProcessed.BlockNumber-lp.backupPollerBlockDelay)
```

You can't subtract `lp.backupPollerBlockDelay` from `lastProcess.BlockNumber` without explicit casting, and having to worry about overflows while casting.  So it ended up being much cleaner if everything that expresses a quantity of blocks is int64